src/Education/XamlGenerator/DiffTranslator.cs

```diff
@@ -109,5 +97,30 @@ private string HighlightLines(List<DiffPiece> lines, StyleResourceNames style, S
             sb.Replace("\r\n", "\n");
             return sb.ToString();
         }
+
+        private void WriteSubPieces(XmlWriter writer, StyleResourceNames style, DiffPiece line)
+        {
+            if (line.SubPieces.Count == 0)
+            {
+                writer.WriteString(line.Text);
+                return;
+            }
+
+            foreach (var subPiece in line.SubPieces)
+            {
+                if (subPiece.Type == ChangeType.Imaginary) { continue; }
```

Comment:
Imaginary changes are a bit strange, I can't quite explain what it means but whenever a subpiece is imaginary its text is null in which case we should ignore it.